tests/phpunit/ApiTest.php

```diff
@@ -0,0 +1,601 @@
+    }
+
+    public function testCreateDatamodel(): void
+    {
+        $api = $this->getApiConnection();
+        $api->login();
+
+        $datamodel = $api->createDatamodel(uniqid('datamodel-'));
+
+        $this->assertNotEmpty($datamodel);
+        $this->assertArrayHasKey('oid', $datamodel);
+
+        $api->deleteDatamodel($datamodel['oid']);
+    }
+
+    public function testDuplicityDatamodel(): void
+    {
+        $api = $this->getApiConnection();
+        $api->login();
+
+        $datamodelName = uniqid('datamodel-');
+
+        $datamodel = $api->createDatamodel($datamodelName);
+
+        try {
+            $api->createDatamodel($datamodelName);
+            $this->fail('Cannot create duplicity datamodel name failed');
+        } catch (UserException $exception) {
+            // phpcs:disable Generic.Files.LineLength
+            $expectedMessage = 'Client error: `POST http://%s:%s/api/v2/datamodels` resulted in a `400 Bad Request` response:
+{"type":"https://errors.sisense.dev/http/general-error","title":"ElasticubeAlreadyExists","status":400,"sub":1011,"detai (truncated...)
+';
+            // phpcs:enable
+            $this->assertEquals(sprintf(
+                $expectedMessage,
+                getenv('SISENSE_HOST'),
+                getenv('SISENSE_PORT')
+            ), $exception->getMessage());
+        }
+        $api->deleteDatamodel($datamodel['oid']);
+    }
+
+    public function testExistsDatamodel(): void
+    {
+        $api = $this->getApiConnection();
+        $api->login();
+
+        $datamodelName = uniqid('datamodel-');
+
+        $createDatamodel = $api->createDatamodel($datamodelName);
+
+        $existsDatamodel = $api->checkDatamodelExists($datamodelName);
+
+        unset($createDatamodel['lastSuccessfulBuildTime']);
+        $this->assertNotNull($createDatamodel);
+        $this->assertNotNull($existsDatamodel);
+        $this->assertEquals($createDatamodel, $existsDatamodel);
+
+        $api->deleteDatamodel($createDatamodel['oid']);
+    }
+
+    public function testInvalidExistsDatamodel(): void
+    {
+        $api = $this->getApiConnection();
+        $api->login();
+
+        $invalidDatamodel = $api->checkDatamodelExists('invalidDatamodel');
+        $this->assertNull($invalidDatamodel);
+    }
+
+    public function testCreateDataset(): void
+    {
+        $api = $this->getApiConnection();
+        $api->login();
+
+        $datamodel = $api->createDatamodel(uniqid('datamodel-'));
+
+        $csvFile = $this->createCsvExampleFile();
+        $sisenseFile = $api->uploadFile(
+            $csvFile->getPath() . '/',
+            $csvFile->getFilename()
+        );
+
+        $dataset = $api->createDataset($datamodel['oid'], uniqid('dataset-'), $sisenseFile, $csvFile->getFilename());
+
+        $this->assertNotEmpty($dataset);
+        $this->assertArrayHasKey('oid', $dataset);
+
+        $api->deleteDataset($datamodel['oid'], $dataset['oid']);
```

Comment:
mozno by to bolo dobre balit do nejakeho `(create) try {} finally {(delete)}` aby sa to vymazalo aj ked test neprejde, ... moze nastat nejaky error na API, alebo neprejde Assert (ten vyhadzuje vzdy exception) ... ina alternativa je v `tearDown` metode nacitat vsetky objekty (datasets, tables, ...) cez API a vsetky zmazat.

Reply:
dal jsem to do finally protože tearDown se pouští vždy a ne vždy to je potřeba 👍 díky